src/weather/tests/weather.service.e2e-spec.ts

```diff
@@ -0,0 +1,37 @@
+import * as request from 'supertest';
+import { Test } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import { WeatherApiResponse } from '../models/weather.model';
+import { WeatherModule } from '../weather.module';
+import { WeatherService } from '../weather.service';
+import { WeatherResponseMock } from '../constants/weather-response.mock';
+
+describe('Weather', () => {
+  let app: INestApplication;
+  const weatherService = {
+    getForecast: (): WeatherApiResponse => WeatherResponseMock,
+  };
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [WeatherModule],
+    })
+      .overrideProvider(WeatherService)
+      .useValue(weatherService)
+      .compile();
+
+    app = moduleRef.createNestApplication();
+    await app.init();
+  });
+
+  it(`/GET weather`, () => {
+    return request(app.getHttpServer())
+      .get('/weather')
+      .expect(200)
+      .expect(WeatherResponseMock);
```

Comment:
Since it's in `PascalCase` I though it was a class 🤔 maybe a `WEATHER_RESPONSE_MOCK` name would be better to show it's a constant of some sort?